File: IT/Interfacce grafiche/Esercizi/C++/esercizio18/esercizio18.cpp
/* file esercizio18.cpp */

#include <QApplication>
#include <QVBoxLayout>
#include <QMenuBar>
#include <QLabel>
#include "esercizio18.h"

Esercizio18::Esercizio18(QWidget *parent) : QMainWindow(parent)
{
	QMenu *fileMenu = menuBar()->addMenu("&File");

	QAction *action1 = new QAction("Testa", this);
	QAction *action2 = new QAction("Croce", this);
	QAction *closeAction = new QAction("Esci", this);

	fileMenu->addAction(action1);
	fileMenu->addAction(action2);
	fileMenu->addAction(closeAction);

	connect(action1, SIGNAL(triggered()), this, SLOT(menu1ClickHandler()));
	connect(action2, SIGNAL(triggered()), this, SLOT(menu2ClickHandler()));
	connect(closeAction, SIGNAL(triggered()), this, SLOT(closeHandler()));

	text = new QTextEdit();

	QWidget *widget = new QWidget(this);
	QVBoxLayout *layout = new QVBoxLayout(widget);
	layout->addWidget(text);

	QLabel *label = new QLabel("Lancia la monetina con il menu a tendina.");
	layout->addWidget(label);

	widget->setLayout(layout);
	this->setCentralWidget(widget);

	this->resize(280, 250);
	this->setWindowTitle("Testa o croce?");
	this->show();
}

void Esercizio18::menu1ClickHandler()
{
	text->append("Testa");
}

void Esercizio18::menu2ClickHandler()
{
	text->append("Croce");
}

void Esercizio18::closeHandler()
{
	QApplication::quit();
}

File: IT/2/Interfacce grafiche/Esercizi/Python/esercizio18.py
# file esercizio18.py

try:
	from tkinter import Frame, Label, Button, Menu, INSERT
	from tkinter.scrolledtext import ScrolledText
except ImportError:
	from Tkinter import Frame, Label, Button, Menu, INSERT
	from ScrolledText import ScrolledText

class Widget(Frame):
	def __init__(self, parent):
		Frame.__init__(self, master=parent, width=280, height=200)
		self.scrolledText = ScrolledText(self)
		self.scrolledText.pack()
		self.pack_propagate(False)
		self.pack()

class Esercizio18(Frame):
	def __init__(self):
		Frame.__init__(self, master=None)

		self.menubar = Menu(self)
		fileMenu = Menu(self.menubar)

		fileMenu.add_command(label="Testa", command=self.menu1ClickHandler)
		fileMenu.add_command(label="Croce", command=self.menu2ClickHandler)
		fileMenu.add_command(label="Esci", command=self.closeHandler)
		self.menubar.add_cascade(label="File", menu=fileMenu)

		self.widget = Widget(self)
		self.label = Label(self, text="Lancia la monetina con il menu a tendina.")
		self.label.pack()

		self.master.config(menu=self.menubar)

		self.pack()
		self.master.wm_title("Testa o croce?")
		self.mainloop()

	def menu1ClickHandler(self):
		self.widget.scrolledText.insert(INSERT, "Testa\n")

	def menu2ClickHandler(self):
		self.widget.scrolledText.insert(INSERT, "Croce\n")

	def closeHandler(self):
		self.master.destroy()

Esercizio18()

